gui/src/controls/control.ts

```diff
@@ -768,6 +768,13 @@ module BABYLON.GUI {
         
             this._fontOffset = Control._GetFontOffset(this._font);
         }
+        
+        public measureForIdeal() {
+            this._measure();
+            
+            this.width = this._currentMeasure.width + 'px';

```

Comment:
Basically for this purpose it shouldn't matter if currentMeasure will be changed (if I get the picture correctly).
The point is to do a 1-time measurement of the control and set its width and height in px, as if it were done externally by the user, in order for the control to be resized according to idealWidth and idealHeight later on.

I created a container and its width and height seem to have been set properly, but I can't say I thoroughly tested all cases, or that I even entirely understand what's going on.